src/sentry/integrations/on_call/spec.py

```diff
@@ -0,0 +1,16 @@
+from sentry.models.notificationaction import ActionService
+
+
+class OnCallSpec:
```

Comment:
i would suggest making this into an abstract base class. @RyanSkonnord has examples for the messaging integrations, and in the future we likely want to do something similar anyway.

i suggest this because you can specify what each oncall integration requires in the abstract base class, and then have each oncall integration inherit from it rather than have a single class.

there is some hardcoded stuff in this class for action_service, but if you had it as an `abstractmethod`, you could then define `action_service` within the pagerduty / opsgenie spec classes that inherit from this one. this way, we keep the definition within each class

something like

```
class OnCallSpec(abc.ABC):
   @abstractmethod
   @property
   def provider_slug(self):
       raise NotImplementedError

   @abstractmethod
   @property
   def action_service(self):
       raise NotImplementedError


class OpsgenieOnCallSpec(OnCallSpec):
   @property
   def provider_slug(self):
       return "opsgenie"

   @property
   def action_service(self):
       return ActionService.OPSGENIE